lldb/source/Core/ModuleList.cpp

```diff
@@ -310,6 +310,14 @@ ModuleList::ModuleList(const ModuleList &rhs) : m_modules(), m_modules_mutex() {
   m_modules = rhs.m_modules;
 }
 
+ModuleList::ModuleList(ModuleList &&rhs) : m_modules(), m_modules_mutex() {
+  std::lock_guard<std::recursive_mutex> lhs_guard(m_modules_mutex);
+  std::lock_guard<std::recursive_mutex> rhs_guard(rhs.m_modules_mutex);
+  m_modules = std::move(rhs.m_modules);
+  // The vector should already be cleared, but that's not guaranteed by the
+  // standard.
+  rhs.Clear();
```

Comment:
According to cppreference, it is guaranteed:

> After the move, other is guaranteed to be [empty()](https://en.cppreference.com/w/cpp/container/vector/empty).

Reply:
That's for the constructor, we're using the assignment operator in this case, which in this case says (https://en.cppreference.com/w/cpp/container/vector/operator%3D):
> other is in a valid but unspecified state afterwards  

We're using the assignment operator because we have to initialize our vector in the constructor's body since we need to lock before that (can't use constructor initializer list).

Reply:
You're right!